src/Search/README_SEARCH.md

```diff
@@ -158,64 +174,127 @@ However, if you're going to apply filters to a query, that means you're going to
 
 ```html
-<form>
-  <input type="text" name="dfParam_query">
-  <input type="hidden" name="dfParam_query_name" value="text_all">
+<form method="GET" action="search.php">
+  <input type="hidden" name="hashid" value="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa">
+  <input type="text" name="query" value="sneakers">
+  <input type="hidden" name="query_name" value="match_and">
   ...
 </form>
 ```
 
+```php
+$params = $client->searchParams($_GET);
+// [
+//     "hashid" => "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+//     "query" => "sneakers",
+//     "query_name" => "match_and"
+//     …
+// ]
+```
+
+You can change the name of the query parameter:
+
+```html
+<form method="GET" action="search.php">
+  <input type="hidden" name="hashid" value="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa">
+  <input type="text" name="text" value="sneakers">
+  <input type="hidden" name="query_name" value="match_and">
+  ...
+</form>
+```
+
+```php
+$params = $client->searchParams($_GET, ["queryParameter" => "text"]);
+// [
+//     "hashid" => "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+//     "query" => "sneakers",
+//     "query_name" => "match_and"
+//     …
+// ]
+```
+
+And you can add a prefix if necessary:
+
+```html
+<form method="GET" action="search.php">
+  <input type="hidden" name="dfParam_hashid" value="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa">
+  <input type="text" name="dfParam_text" value="sneakers">
+  <input type="hidden" name="dfParam_query_name" value="match_and">
+  ...
+</form>
+```
+
+```php
+$params = $client->searchParams($_GET, ["prefix" => "dfParam_", "queryParameter" => "text"]);
+// [
+//     "hashid" => "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+//     "query" => "sneakers",
+//     "query_name" => "match_and",
+//     …
+// ]
+```
+
 In short:
 
 - You make a query to Doofinder either with filters or not. You don't need to specify `query_name`. Doofinder will find the more suitable `query_name`.
-- Once you got the results, you can use `$results->getProperty('query_name')` to know which `query_name` was the one Doofinder chose.
+- Once you got the results, you can use `$results->getProperty('query_name')` to know which `query_name` was the one Doofinder chose in a specific set of results or use `$client->getSearchParam("query_name")` to get the query name from the latest call.
 - If you want to make further filtering on those search results, you should instruct Doofinder to use the same `query_name` you got from the first search results.
 - Each time you do any new query, don't specify `query_name`. Let Doofinder find the best.
-- **Warning:** don't try to figure out a `query_name` on your own. query names may change in the future. You can always count on `$results->getParameter('query_name')` to get the `query_name` that led to those `$results`
+- **Warning:** don't try to figure out a `query_name` on your own. Query names may change in the future. You can always count on `$results->getParameter('query_name')` to get the `query_name` that led to those `$results`.
```

Comment:
shouldn't it be `$client->getSearchParam("query_name")` ?

Reply:
It can actually be both because I didn't change the Results class. But I'm going to tell it. Thanks!